### ISSUE TITLE
Gitlab asks for git user password when pushing a repo.

### ISSUE THREAD

**Issue by arnaudchenyensu** [sameersbn/docker-gitlab#34]
When I push a repo, Gitlab asks me the password for the git user:
`git@gitlab.mydomain.com's password:`

When I try:

```
> ssh -Tvvv git@ns24652.ip-91-121-24.eu
...
debug1: Found key in /Users/arnaud/.ssh/known_hosts:18
debug2: bits set: 505/1024
debug1: ssh_rsa_verify: signature correct
debug2: kex_derive_keys
debug2: set_newkeys: mode 1
debug1: SSH2_MSG_NEWKEYS sent
debug1: expecting SSH2_MSG_NEWKEYS
debug2: set_newkeys: mode 0
debug1: SSH2_MSG_NEWKEYS received
debug1: Roaming not allowed by server
debug1: SSH2_MSG_SERVICE_REQUEST sent
debug2: service_accept: ssh-userauth
debug1: SSH2_MSG_SERVICE_ACCEPT received
debug2: key: /Users/arnaud/.ssh/id_rsa (0x7fbf42500380),
debug2: key: /Users/arnaud/.ssh/id_dsa (0x7fbf42500220),
debug1: Authentications that can continue: publickey,password
debug3: start over, passed a different list publickey,password
debug3: preferred publickey,keyboard-interactive,password
debug3: authmethod_lookup publickey
debug3: remaining preferred: keyboard-interactive,password
debug3: authmethod_is_enabled publickey
debug1: Next authentication method: publickey
debug1: Offering RSA public key: /Users/arnaud/.ssh/id_rsa
debug3: send_pubkey_test
debug2: we sent a publickey packet, wait for reply
debug1: Authentications that can continue: publickey,password
debug1: Offering DSA public key: /Users/arnaud/.ssh/id_dsa
debug3: send_pubkey_test
debug2: we sent a publickey packet, wait for reply
debug1: Authentications that can continue: publickey,password
debug2: we did not send a packet, disable method
debug3: authmethod_lookup password
debug3: remaining preferred: ,password
debug3: authmethod_is_enabled password
debug1: Next authentication method: password
git@gitlab.mydomain.com's password:
```

I tried the solution at issue #26 and other solutions on internet but nothing worked.

**Comment by mbannert** [comment]
@sameersbn **You cannot ssh to the gitlab server as git user** thanks for stating this!! I wasted a lot of time cause I tried to clone using the git user. Pull / push seems to work. Gitlab seems really nice to me know.

**Comment by sameersbn** [comment]
@mbannert please note, using this image you should be able to clone git repos via HTTP as well as SSH.

**Comment by 18601673727** [comment]
Is this issue got solved? I met this issue when i do:
```txt
git clone git@git.mydomain.com:myaccount/xxx.git
Cloning into 'xxx'...
git@git.mydomain.com's password:
Permission denied, please try again.  # I don't know this password, so pressed "enter"
```
I used omnibus version of Gitlab before and SSH is fine without password required. 

Btw, there is a load balancer in front of `docker-gitlab` and configured like this:
```yml
  ports:
    - "10080:80"
    - "10022:22"
  environment:
    - GITLAB_HOST=git.mydomain.com
    - GITLAB_PORT=443
    - GITLAB_SSH_PORT=22
    - GITLAB_HTTPS=true
    - SSL_SELF_SIGNED=false # Using Letsencrypt
```

The web UI is fine and HTTPS ways good, SSH pubkey added via web UI.

Any help would be appreciate. 😄

**Comment by bfdill** [comment]
I am in the exact same boat as @18601673727.  Any pointers?

**Comment by sameersbn** [comment]
@bfdill can you create a new issue with details such as the sample docker-compose.yml file, docker version, host os in use, whether boot2docker, etc is in use?

**Comment by sameersbn** [comment]
I noticed that you have published `10022:22` while you have `GITLAB_SSH_PORT=22`. As a result the clone url may be wrong. To me it appears that the git clone is connecting to the host ssh server, not the containers. You need to set  `GITLAB_SSH_PORT=10022` and then use the new clone url which will look something like this `ssh://git@git.mydomain.com:10022/my-account/xxx.git`.

If you want to have clone urls like `git@git.mydomain.com:myaccount/xxx.git`, you will first need to configure the host SSH daemon to listen on another port. In the docker-compose.yml change `10022:22` to `22:22` and set `GITLAB_SSH_PORT=22`

**Comment by 18601673727** [comment]
@sameersbn You're solution perfectly solved this issue, thanks!

**Comment by elyman** [comment]
0
down vote
I had the same problem, I spent a lot of time searching! I had the idea to use Eclipse to import the project from GitLab. Once the project is imported correctly. I made the comparison between the configuration of : project's Git ripository that I imported into Eclispe, ("in Eclipse", Git Repository, in myprojectRepo / Working Directory / .git / config) and one that is made in .git / config, there i wanted to push my project with git: git push ... and asked me for a password. Surpise: The remote does not have the same in both cases. I handed the same as that in eclipse and everything works.
